src/ReportsSync.php

```diff
@@ -9,197 +9,40 @@
-use \Automattic\WooCommerce\Admin\API\Reports\Products\DataStore as ProductsDataStore;
-use \Automattic\WooCommerce\Admin\API\Reports\Taxes\DataStore as TaxesDataStore;
-use \Automattic\WooCommerce\Admin\API\Reports\Cache as ReportsCache;
+use Automattic\WooCommerce\Admin\Schedulers\CustomersScheduler;
+use Automattic\WooCommerce\Admin\Schedulers\OrdersScheduler;
 
 /**
  * ReportsSync Class.
  */
 class ReportsSync {
-	/**
-	 * Action hook for reducing a range of batches down to single actions.
-	 */
-	const QUEUE_BATCH_ACTION = 'wc-admin_queue_batches';
-
-	/**
-	 * Action hook for queuing an action after another is complete.
-	 */
-	const QUEUE_DEPENDENT_ACTION = 'wc-admin_queue_dependent_action';
-
-	/**
-	 * Action hook for importing a batch of customers.
-	 */
-	const CUSTOMERS_IMPORT_BATCH_ACTION = 'wc-admin_import_customers_batch';
-
-	/**
-	 * Action hook for initializing the orders lookup batch creation.
-	 */
-	const CUSTOMERS_DELETE_BATCH_INIT = 'wc-admin_delete_customers_batch_init';
-
-	/**
-	 * Action hook for deleting a batch of customers.
-	 */
-	const CUSTOMERS_DELETE_BATCH_ACTION = 'wc-admin_delete_customers_batch';
-
-	/**
-	 * Action hook for importing a batch of orders.
-	 */
-	const ORDERS_IMPORT_BATCH_ACTION = 'wc-admin_import_orders_batch';
-
-	/**
-	 * Action hook for initializing the orders lookup batch creation.
-	 */
-	const ORDERS_IMPORT_BATCH_INIT = 'wc-admin_orders_lookup_import_batch_init';
-
-	/**
-	 * Action hook for initializing the orders lookup batch deletion.
-	 */
-	const ORDERS_DELETE_BATCH_INIT = 'wc-admin_orders_lookup_delete_batch_init';
-
-	/**
-	 * Action hook for deleting a batch of orders.
-	 */
-	const ORDERS_DELETE_BATCH_ACTION = 'wc-admin_delete_orders_batch';
-
-	/**
-	 * Action hook for importing a single customer.
-	 */
-	const SINGLE_CUSTOMER_IMPORT_ACTION = 'wc-admin_import_customer';
-
-	/**
-	 * Action hook for importing a single order.
-	 */
-	const SINGLE_ORDER_IMPORT_ACTION = 'wc-admin_import_order';
-
-	/**
-	 * Action scheduler group.
-	 */
-	const QUEUE_GROUP = 'wc-admin-data';
-
-	/**
-	 * Queue instance.
-	 *
-	 * @var WC_Queue_Interface
-	 */
-	protected static $queue = null;
-
-	/**
-	 * Get queue instance.
-	 *
-	 * @return WC_Queue_Interface
-	 */
-	public static function queue() {
-		if ( is_null( self::$queue ) ) {
-			self::$queue = WC()->queue();
-		}
-
-		return self::$queue;
-	}
-
-	/**
-	 * Set queue instance.
-	 *
-	 * @param WC_Queue_Interface $queue Queue instance.
-	 */
-	public static function set_queue( $queue ) {
-		self::$queue = $queue;
-	}
-
 	/**
 	 * Hook in sync methods.
 	 */
 	public static function init() {
-		// Initialize syncing hooks.
-		add_action( 'wp_loaded', array( __CLASS__, 'customers_lookup_update_init' ) );
-		add_action( 'wp_loaded', array( __CLASS__, 'orders_lookup_update_init' ) );
+		// Initialize scheduler hooks.
+		foreach ( self::get_schedulers() as $scheduler ) {
+			$scheduler::init();
```

Comment:
Since there's a filter inside `get_schedulers()`, should we check to make sure `init` is callable here?

Reply:
Resolved via https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/e7fd4e813f7c3618fd07d33243930e801c008a9e